### PR TITLE
feat: Add ANR detection for main process

### PR DESCRIPTION
**Pull request by timfish**
Ref #729

This PR adds ANR detection for the Electron main process:

```ts
import { init, enableAnrDetection } from '@sentry/electron';
 
init({ dsn: "__DSN__" });
 
// Electron v28+ with ESM
await enableAnrDetection({ mainProcess: { captureStackTrace: true }});
runApp();

// with CJS
enableAnrDetection({ mainProcess: { captureStackTrace: true }}).then(() => {
   runApp();
});
```

Should we keep the main/renderer config separate like this?

### CODE REVIEW

File: src/main/anr.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    options.mainProcess = options.mainProcess || {};
+
+    // We need to override the entryScript option to make it work with Electron which doesn't get passed a script in
+    // the process.argv when the app is packaged
+    if (options.mainProcess.entryScript === undefined) {
+      options.mainProcess.entryScript = app.getAppPath();
+    }
+
+    return enableNodeAnrDetection(options.mainProcess);
```
Comment (AbhiPrasad) [inline src/main/anr.ts:52]: let's avoid mutating the original options object. We removed this pattern in the js codebase.

**Review by AbhiPrasad — APPROVED** [review]
Looks great!

**Comment by AlbertInRC** [comment]
Hi @tim @AbhiPrasad , will the ANR events impact the Crash Free Session rate?

**Comment by timfish** [comment]
> will the ANR events impact the Crash Free Session rate?

No, not currently.

**Pull request merged**